third-party/rsutils/include/rsutils/string/slice.h

```diff
@@ -0,0 +1,86 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <ostream>
+#include <cassert>
+#include <cstring>  // std::strlen
+
+
+namespace rsutils {
+namespace string {
+
+
+// Same as std::string, except not null-terminated, and does not manage its own memory! Similar to the C++17
+// std::string_view, except it can be expanded to be non-const, too.
+//
+// This is meant to point into an existing string and have a short life-time. If the underlying memory is removed, e.g.:
+//      slice foo()
+//      {
+//          std::string bar = "haha";
+//          return slice(bar);
+//      }
+// Not good!
+//
+// This can come in very useful when wanting to break a string into parts but without incurring allocations or copying,
+// or changing of the original string contents (like strtok does), or when the original memory is not null-terminated.
+//
+class slice
+{
+public:
+    typedef char const * const_iterator;
+    typedef std::string::size_type size_type;
+    typedef std::string::value_type value_type;
+
+private:
+    const_iterator _begin, _end;
+
+public:
+    slice( const_iterator begin, const_iterator end )
```

Comment:
It is better to leave a blank line between functions.
Fix if you'll have some other changes to this file.